src/Database/index.js

```diff
@@ -178,12 +178,14 @@ class Database {
    * ```
    */
   beginTransaction () {
+    if (this._globalTrx) {
+      return new Promise((resolve, reject) => {
+        this._globalTrx.transaction(resolve).catch(() => {})
+      })
+    }
+
     return new Promise((resolve, reject) => {
-      this
-        .knex
-        .transaction(function (trx) {
-          resolve(trx)
-        }).catch(() => {})
+      this.knex.transaction(resolve).catch(() => {})
```

Comment:
May be off topic, but why are we catching without rejecting? Not ever rejecting just causes the promise to be left pending indefinitely. In a web app this can cause a server to crash if enough transactions fail (e.g. database is overloaded). A preferable outcome would be to reject so that the server can handle the result gracefully.